mu4/scenes/instruments/view/instrumentlistmodel.cpp

```diff
@@ -109,23 +99,49 @@ QVariantList InstrumentListModel::groups()
 
 QVariantList InstrumentListModel::instruments()
 {
-    if (!m_inited) {
-        return QVariantList();
-    }
+    QVariantHash availableInstruments;
 
-    QVariantList result;
     for (const InstrumentTemplate& instrument: m_instrumentsMeta.instrumentTemplates) {
-        if (instrument.groupId != m_selectedGroupId || !instrument.genreIds.contains(m_selectedFamilyId)) {
+        if (instrument.groupId != m_selectedGroupId) {
             continue;
         }
 
-        QVariantMap obj;
-        obj["name"] = instrument.trackName;
+        if (!isInstrumentAccepted(instrument)) {
+            continue;
+        }
 
-        result << obj;
+        QVariantList instrumentTranspositions;
+        QString instrumentId = instrument.id;
+        QString instrumentName = instrument.trackName;
+
+        Transposition _transposition = transposition(instrument.id);
+        if (_transposition.isValid()) {
+            instrumentId = instrumentId.replace(_transposition.id, "");
+            instrumentName = instrumentName.replace(_transposition.name + " ", "")
+                             .replace(" in " + _transposition.name, "");
+
+            if (availableInstruments.contains(instrumentId)) {
+                instrumentTranspositions = availableInstruments[instrumentId].toMap().value("transpositions").toList();
+            }
+
+            QVariantMap obj;
+            obj["id"] = _transposition.id;
```

Comment:
These keys ("id", "name") can be found in several places. Better to add constants: static const QString KEY_ID("id")